src/DataViews/FunctionsDataView.cpp

```diff
@@ -244,78 +242,55 @@ void FunctionsDataView::OnContextMenu(const std::string& action, int menu_index,
-
-    bool match = true;
-
-    for (std::string& filter_token : m_FilterTokens) {
-      if (name.find(filter_token) == std::string::npos) {
-        match = false;
-        break;
-      }
-    }
+  const size_t number_of_threads_available = thread_pool_->GetPoolSize();
 
-    if (match) {
-      indices.push_back(i);
-    }
-  }
+  constexpr size_t kMinimumNumberOfFunctionsPerThread = 64;
+  const size_t number_of_functions_per_thread =
+      std::max(kMinimumNumberOfFunctionsPerThread, functions_.size() / number_of_threads_available);
+  const size_t number_of_threads_needed =
+      functions_.size() / number_of_functions_per_thread +
+      ((functions_.size() % number_of_functions_per_thread) > 0 ? 1 : 0);
 
-  indices_ = std::move(indices);
-#endif
-}
+  std::vector<orbit_base::Future<std::vector<uint64_t>>> filtered_indices_per_thread;
+  filtered_indices_per_thread.reserve(number_of_threads_needed);
 
-void FunctionsDataView::ParallelFilter() {
-#ifdef _WIN32
-  const std::vector<const FunctionInfo*>& functions(functions_);
-  const auto prio = oqpi::task_priority::normal;
-  auto numWorkers = oqpi::default_helpers::scheduler().workersCount(prio);
-  // int numWorkers = oqpi::thread::hardware_concurrency();
-  std::vector<std::vector<int>> indicesArray;
-  indicesArray.resize(numWorkers);
-
-  oqpi::default_helpers::parallel_for(
-      "FunctionsDataViewParallelFor", functions.size(),
-      [&](int32_t a_BlockIndex, int32_t a_ElementIndex) {
-        std::vector<int>& result = indicesArray[a_BlockIndex];
-        const std::string& name = absl::AsciiStrToLower(
-            orbit_client_data::function_utils::GetDisplayName(*functions[a_ElementIndex]));
-        const std::string& module =
-            orbit_client_data::function_utils::GetLoadedModuleName(*functions[a_ElementIndex]);
-
-        for (std::string& filterToken : m_FilterTokens) {
-          if (name.find(filterToken) == std::string::npos &&
-              module.find(filterToken) == std::string::npos) {
-            return;
-          }
-        }
+  for (size_t thread_idx = 0; thread_idx < number_of_threads_needed; ++thread_idx) {
+    const size_t begin = thread_idx * number_of_functions_per_thread;
+    const size_t end =
+        std::min((thread_idx + 1) * number_of_functions_per_thread, functions_.size());
 
-        result.push_back(a_ElementIndex);
-      });
+    filtered_indices_per_thread.emplace_back(thread_pool_->Schedule([begin, end, this]() {
+      std::vector<uint64_t> indices_of_matches;
 
-  std::set<int> indicesSet;
-  for (std::vector<int>& results : indicesArray) {
-    for (int index : results) {
-      indicesSet.insert(index);
-    }
+      for (size_t index = begin; index < end; ++index) {
+        const FunctionInfo* function = functions_[index];
+        std::string name =
+            absl::AsciiStrToLower(orbit_client_data::function_utils::GetDisplayName(*function)) +
```

Comment:
Shouldn't there at least be a space between name and module? What happens if the name is `foo`, the module is `mod`, and I look for `oom`?

Reply:
Actually could you add a test for this?

Reply:
I haven't changed the logic as I wanted to keep the behaviour. This is copy-and-pasted from the original implementation.
But yes, I can add a test.

Reply:
Ah, I was looking at `ParallelFilter` and the implementation there is different, so the behavior was different between Linux and Windows.

Reply:
Yes, that surprised me as well. But that's actually a good point. Maybe we should keep the Windows behaviour. I will have a look!

Reply:
Changed to the Windows behaviour and added tight tests.